pkg/api/resource/quantity.go

```diff
@@ -615,6 +620,15 @@ func (q *Quantity) String() string {
 	return q.s
 }
 
+func (q *Quantity) Len() int {
+	if len(q.s) == 0 {
+		result := make([]byte, 0, int64QuantityExpectedBytes)
+		number, suffix := q.CanonicalizeBytes(result)
```

Comment:
@smarterclayton - I tried to go step further and create a sibling method for CanonicalizeBytesLen, but:
- it requires creating a bunch of other sibling methods (i.e. ~10 of them), which makes the code much less clear/readable
- there is still one allocating (though we would be allocating ~3x less bytes).

Do you think it would be worth doing?